testtools/matchers/_warnings.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2009-2016 testtools developers. See LICENSE for details.
+
+__all__ = [
+    'Warnings',
+    'warning_message',
+    'is_deprecated']
+
+import warnings
+
+from ._basic import Is
+from ._const import Always
+from ._datastructures import MatchesListwise, MatchesStructure
+from ._higherorder import (
+    AfterPreprocessing,
+    Annotate,
+    MatchesAll,
+    Not,
+    )
+from ._impl import Mismatch
+
+
+def warning_message(category_type, message=None, filename=None, lineno=None,
```

Comment:
IMO, `raises` and `Raises` are an anti-pattern. As a testtools user, I can never remember which one does what (although that might have more to do with having two objects with the same name and different case). As @jml commented on the MP, We recently decided to export functions as CamelCase names for the sake of consistency. That decision isn't perfect, but I think in this case there's no good answer, and 'CamelCase'ing everything is probably the best of a bunch of bad options.